src/components/button/button.spec.tsx

```diff
@@ -121,234 +114,148 @@ describe("Button", () => {
+    const wrapper = mount(
+      <Button iconType="filter" iconPosition="before">
+        Filter
+      </Button>
+    );
+    const button = wrapper.find("button");
+    expect(button.childAt(0).find(Icon).exists()).toBeTruthy();
+    expect(button.childAt(1).text()).toEqual("Filter");
+  });
 
-    describe.each<ButtonIconPosition>(["before", "after"])(
-      'when position is set to "%s"',
-      (position) => {
-        describe.each<ButtonTypes>(BUTTON_VARIANTS)(
-          "and the button type is %s",
-          (buttonType) => {
-            beforeEach(() => {
-              wrapper = render({
-                children: "foo",
-                iconType: "filter",
-                iconPosition: position,
-                buttonType,
-              }).dive();
-            });
-
-            it("contains an Icon", () => {
-              const assertion =
-                wrapper.find(Icon).exists() &&
-                wrapper.find(Icon).props().type === "filter";
-              expect(assertion).toEqual(true);
-            });
-          }
-        );
-      }
+  it("icon is rendered after text when iconPosition is 'before' and iconType is specified", () => {
+    const wrapper = mount(
+      <Button iconType="filter" iconPosition="after">
+        Filter
+      </Button>
     );
+    const button = wrapper.find("button");
+    expect(button.childAt(0).text()).toEqual("Filter");
+    expect(button.childAt(1).find(Icon).exists()).toBeTruthy();
   });
 
-  describe("When icon type is specified and button has no children", () => {
-    describe.each(BUTTON_VARIANTS)(
-      "and the button type is %s",
-      (buttonType) => {
-        beforeEach(() => {
-          wrapper = render({
-            iconType: "filter",
-            buttonType,
-          }).dive();
-        });
+  it("renders with correct border radius", () => {
+    const wrapper = mount(<Button>foo</Button>);
+    assertStyleMatch({ borderRadius: "var(--borderRadius400)" }, wrapper);
+  });
 
-        it("contains an Icon", () => {
-          const assertion =
-            wrapper.find(Icon).exists() &&
-            wrapper.find(Icon).props().type === "filter";
-          expect(assertion).toEqual(true);
-        });
-      }
+  it("error is thrown when there are no iconType and no children passed", () => {
+    const consoleSpy = jest
+      .spyOn(console, "error")
+      .mockImplementation(() => {});
+
+    expect(() => mount(<Button />)).toThrow(
+      "Either prop `iconType` must be defined or this node must have children."
     );
-  });
 
-  describe("when there are no props passed except children", () => {
-    it('matches the expect styles for a default button with variants "secondary" and "medium"', () => {
-      wrapper = render({ children: "foo" }, TestRenderer.create).toJSON();
+    consoleSpy.mockRestore();
+  });
+
+  it.each([
+    [
+      "primary",
+      "var(--colorsActionDisabled500)",
+      "transparent",
+      "var(--colorsActionMajorYin030)",
```

Comment:
Suggestion: as every `color` is the same value we could probably simplify this array by removing it and just asserting `color: var(--colorsActionMajorYin030)` in the test